src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1742,16 +1726,12 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     void ControlCore::ClearSearch()
     {
-        // nothing to clear if there's no results
-        if (_searcher.GetCurrent())
-        {
-            const auto lock = _terminal->LockForWriting();
-            _terminal->SetSearchHighlights({});
-            _terminal->SetSearchHighlightFocused({});
-            _renderer->TriggerSearchHighlight(_searcher.Results());
-            _searcher = {};
-            _cachedSearchResultRows = {};
-        }
+        const auto lock = _terminal->LockForWriting();
+        _terminal->SetSearchHighlights({});
+        _terminal->SetSearchHighlightFocused({});
+        _renderer->TriggerSearchHighlight(_searcher.Results());
+        _searcher = {};
+        _cachedSearchResultRows = {};

```

Comment:
We can't check `_searcher.GetCurrent()` here because otherwise we don't reset the `_lastMutationId` of a `Searcher` that found nothing. This is relevant in case you open the search box (the Ctrl+Shift+F one), enter something that doesn't exist, close it, and reopen it. If it gets closed and doesn't get reset here, then reopening it won't trigger a search otherwise.